File: catkin_ws/src/ssr_pkg/src/yao_node.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

int main(int argc, char *argv[])
{    
    ros::init(argc, argv, "yao_node");
    // 创建ROS节点句柄，用于与ROS系统通信，管理节点资源
    ros::NodeHandle nh;
    printf("ros init success !\n");

    // 创建发布者，用于发布消息，参数为: nh.advertise<std_msgs::消息类型>(话题名称, 缓存队列大小)
    ros::Publisher pub = nh.advertise<std_msgs::String>("yao_topic", 10);
    
    ros::Rate loop_rate(1.0); // 1Hz

    while(ros::ok())
    {    
        std_msgs::String msg;
        msg.data = "yao: test";
        pub.publish(msg);
        
        printf("sleep 1s ...\n");
        loop_rate.sleep(); // 暂停1秒
    }
    return 0;
}

File: catkin_py/src/py_ssr_pkg/scripts/yao_node.py
#!/usr/bin/env python3
#coding=utf-8

import rospy
from std_msgs.msg import String

if __name__ == '__main__':
    rospy.init_node('yao_topic_py')
    rospy.logwarn('[yao_topic_py] init_node success !')

    pub = rospy.Publisher('yao_topic_py', String, queue_size=10)
    rospy.logwarn('[yao_topic_py] publisher success !')

    rate = rospy.Rate(1)

    while not rospy.is_shutdown():
        msg = String()
        msg.data = '[yao_topic_py] hello world'
        pub.publish(msg)
        rospy.logwarn('[yao_topic_py] publish msg success !')
        rate.sleep()